File: Avaliacao Atleta - Aula 22.04/Principal.java
package Atividade;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.HashSet;
import java.util.Scanner;


public class Principal {

	public static void listarNomes (ArrayList<Atleta> lista) {
		if (!lista.isEmpty()) {
		lista.sort((a1, a2)->((String)a1.getNome()).compareTo(a2.getNome()));
		for (Atleta item : lista){
			System.out.println(item);
		}
	}
		else {
			System.out.println("Nenhum atleta cadastrado!");
		}
}
	
	public static void listarPontuacao (ArrayList<Atleta> lista) {
		if (!lista.isEmpty()) {
			lista.sort((a2, a1)->((String)a2.getPontuacao()).compareTo(a1.getPontuacao()));
			for (Atleta pontos : lista){
				System.out.println(pontos);	
			}
		}
		else {
			System.out.println("Nenhum atleta cadastrado!");
		}
	}
	
	public static void cadastrar (HashSet<Atleta> hashAtleta, Atleta jogador , String nome, String apelido, String fone, String dataNascimento, String pontuacao) {
		System.out.println("CADASTRANDO ATLETA: ");
		jogador = new Atleta (nome, apelido, fone, dataNascimento, pontuacao);
		if (!hashAtleta.add(jogador)) {
			System.out.println("Atleta ja cadastrado");
		}
	
	}
	
	
	public static void procurarNome (HashSet<Atleta> hashAtleta, String procuraNome, Atleta jogador) {
		if (jogador.getNome().equals(procuraNome)) {
			System.out.println(procuraNome +" Encontrado! ");
		}
	}
	
	public static void main(String[] args) {
		Scanner teclado = new Scanner (System.in);
		int op;
		int opInterna;
		HashSet<Atleta> hashAtleta = new HashSet<Atleta>();
		ArrayList<Atleta> lista = new ArrayList<Atleta>();
		Atleta jogador = new Atleta();
		String nome, apelido, fone, dataNascimento, pontuacao;
		String procuraNome;
		do {
			System.out.println("Digite uma opcao de acordo com o menu: ");
			System.out.println("1- Listar Atleta (Nome ou pontuacao)");
			System.out.println("2- Cadastrar Atleta");
			System.out.println("3- Procurar Atleta (nome ou apelido)");
			System.out.println("4- Remover atleta (nome ou apelido)");
			op = teclado.nextInt();
			teclado.nextLine();
			
			switch (op) {
			case 1:
				System.out.println("Quer ver por 1- nome ou 2- pontuacao? ");
				opInterna = teclado.nextInt();
				teclado.nextLine();
				lista.clear();
				
				if (opInterna==1) {
					 lista.addAll(hashAtleta);
				     listarNomes(lista);
				}
				if(opInterna==2) {
					lista.addAll(hashAtleta);
					listarPontuacao(lista);
				}
			break;
				
			case 2:
				System.out.println("CADASTRANDO ATLETA: ");
				System.out.println("Digite o nome: ");
				nome = teclado.nextLine();
				System.out.println("Digite o apelido: ");
				apelido = teclado.nextLine();
				System.out.println("Digite o fone: ");
				fone = teclado.nextLine();
				System.out.println("Digite o Data de Nascimento: ");
				dataNascimento = teclado.nextLine();
				System.out.println("Digite a Pontuacao: ");
				pontuacao = teclado.nextLine();
				cadastrar (hashAtleta, jogador, nome, apelido, fone, dataNascimento,pontuacao);
			
				break;
				
				
			case 3:
				System.out.println("Quer ver por 1- Nome ou 2- Apelido? ");
				opInterna = teclado.nextInt();
				teclado.nextLine();
				
				if (opInterna==1) {
					System.out.println("Digite o nome a ser procurado: ");
					procuraNome = teclado.nextLine();
				}
				
				if (opInterna==1) {
					System.out.println("Digite o nome a ser procurado: ");
					procuraNome = teclado.nextLine();
				}
				
				break;
				
			case 4:
				System.out.println("Encerrando programa...");		
				break;
				
			default: 
				System.out.println("Tecla invalida, Digite novamente!");
			}
			
		}while (true);
	}
	}

File: Avaliacao Atleta - Aula 22.04/Avaliacao Python - 29.04/principal.py
from atleta import Atleta


def listar_nomes(lista):
    if lista:
        lista.sort(key=lambda atleta: atleta.nome)
        for item in lista:
            print(item)
    else:
        print("Nenhum atleta cadastrado!")


def listar_pontuacao(lista):
    if lista:
        lista.sort(key=lambda atleta: atleta.pontuacao)
        for pontos in lista:
            print(pontos)
    else:
        print("Nenhum atleta cadastrado!")


def cadastrar(hash_atleta, nome, apelido, fone, data_nascimento, pontuacao):
    print("CADASTRANDO ATLETA: ")
    jogador = Atleta(nome, apelido, fone, data_nascimento, pontuacao)
    if jogador not in hash_atleta:
        hash_atleta.add(jogador)
    else:
        print("Atleta já cadastrado")


def procurar_nome(hash_atleta, procura_nome):
    encontrado = False
    for jogador in hash_atleta:
        if jogador.nome == procura_nome:
            encontrado = True
            print(f"{procura_nome} encontrado!")
            break
    if not encontrado:
        print("Atleta não encontrado.")


def procurar_apelido(hash_atleta, procura_apelido):
    encontrado = False
    for jogador in hash_atleta:
        if jogador.apelido == procura_apelido:
            encontrado = True
            print(f"{procura_apelido} encontrado!")
            break
    if not encontrado:
        print("Atleta não encontrado.")


def remover_nome(hash_atleta, nome):
    removido = False
    for jogador in hash_atleta.copy():
        if jogador.nome == nome:
            hash_atleta.remove(jogador)
            removido = True
            print(f"Atleta removido: {jogador}")
            break
    if not removido:
        print("Atleta não encontrado.")


def remover_apelido(hash_atleta, apelido):
    removido = False
    for jogador in hash_atleta.copy():
        if jogador.apelido == apelido:
            hash_atleta.remove(jogador)
            removido = True
            print(f"Atleta removido: {jogador}")
            break
    if not removido:
        print("Atleta não encontrado.")


def main():
    hash_atleta = set()
    
    while True:
        print("\nDigite uma opção de acordo com o menu:\n")
        print("1- Listar Atletas (Nome ou pontuação)")
        print("2- Cadastrar Atleta")
        print("3- Procurar Atleta (Nome ou apelido)")
        print("4- Remover Atleta (Nome ou apelido)\n")

        op = int(input())
        
        if op == 1:
            print("Quer ver por 1- Nome ou 2- Pontuação? ")
            op_interna = int(input())
            if op_interna == 1:
                lista = list(hash_atleta)
                listar_nomes(lista)
            elif op_interna == 2:
                lista = list(hash_atleta)
                listar_pontuacao(lista)
        elif op == 2:
            print("CADASTRANDO ATLETA: ")
            nome = input("Digite o nome: ")
            apelido = input("Digite o apelido: ")
            fone = input("Digite o fone: ")
            data_nascimento = input("Digite a Data de Nascimento: ")
            pontuacao = input("Digite a Pontuação: ")
            cadastrar(hash_atleta, nome, apelido, fone, data_nascimento, pontuacao)
        elif op == 3:
            print("Quer procurar por 1- Nome ou 2- Apelido? ")
            op_interna = int(input())
            if op_interna == 1:
                procura_nome = input("Digite o Nome a ser procurado: ")
                procurar_nome(hash_atleta, procura_nome)
            elif op_interna == 2:
                procura_apelido = input("Digite o Apelido a ser procurado: ")
                procurar_apelido(hash_atleta, procura_apelido)
        elif op == 4:
            print("Quer remover por 1- Nome ou 2- Apelido? ")
            op_interna = int(input())
            if op_interna == 1:
                remove_nome = input("Digite o Nome a ser removido: ")
                remover_nome(hash_atleta, remove_nome)
            elif op_interna == 2:
                remove_apelido = input("Digite o Apelido a ser removido: ")
                remover_apelido(hash_atleta, remove_apelido)
        elif 5:
            print("Encerrando o programa...")
            return
        
        else:
            print("Tecla inválida. Digite novamente!")


if __name__ == "__main__":
    main()
